tests/e2e/tests/istioctl/istioctl_test.go

```diff
@@ -0,0 +1,355 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+
+	"istio.io/istio/tests/e2e/framework"
+	"istio.io/istio/tests/e2e/util"
+)
+
+const (
+	rulesDir = "tests/apps/istioctl/input"
+
+	// How many retries when waiting for CRUD operation to succeed (database not instantly consistent)
+	maxRetries = 5
+)
+
+var (
+	tc        *testConfig
+	ruleFiles = []string{
+		"route-rule-101.yaml", // An invalid rule
+
+		"route-rule-100-delay-7s.yaml",
+		"route-rule-100-delay-700s.yaml",
+		"route-rule-100-delay-invalid.yaml", // invalid rule
+
+		"mixer-rule-ratings-ratelimit-5per_s.yaml",
+		"mixer-rule-ratings-ratelimit-invalid.yaml",
+	}
+)
+
+type testConfig struct {
+	*framework.CommonConfig
+	gateway  string
+	rulesDir string
+}
+
+type ruleFileAndCreateResponse struct {
+	ruleFile string
+	response *regexp.Regexp
+}
+
+func (t *testConfig) Setup() error {
+	t.gateway = "http://" + tc.Kube.Ingress
+
+	// Customize rule yaml files, replace with actual namespace, and copy to /tmp/<id>
+	for _, rule := range ruleFiles {
+		src := util.GetResourcePath(filepath.Join(rulesDir, rule))
+		dest := filepath.Join(t.rulesDir, rule)
+		orig, err := ioutil.ReadFile(src)
+		if err != nil {
+			glog.Errorf("Failed to read original rule file %s", src)
+			return err
+		}
+		content := string(orig)
+		content = strings.Replace(content, "default", t.Kube.Namespace, -1)
+		err = ioutil.WriteFile(dest, []byte(content), 0600)
+		if err != nil {
+			glog.Errorf("Failed to write into new rule file %s", dest)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *testConfig) Teardown() error {
+	return nil
+}
+
+func waitForRule(ruleKey string, retries int, durBetweenAttempts time.Duration) error {
+
+	for retry := 0; retry < retries; retry++ {
+		time.Sleep(durBetweenAttempts)
+		var output string
+		var err error
+		if output, err = tc.Kube.Istioctl.RunWithOutput([]string{"get", "route-rules"},
```

Comment:
The following would be fewer lines of total code, avoid predeclaring vars, and avoid the line wrap on the important. 

```golang
output, err := tc.Kube.Istioctl.RunWithOutput([]string{"get", "route-rules"},  "", "", false)
if err != nil {
    return err
}
```